src/main/java/io/deephaven/csv/CsvSpecs.java

```diff
@@ -170,6 +205,24 @@ public static Builder builder() {
         return ImmutableCsvSpecs.builder();
     }
 
+    /**
+     * Validates the {@link CsvSpecs}.
+     */
+    @Check
+    protected void check() {
+        // To be friendly, we report all the problems we find at once.
```

Comment:
I think immutables library will be friendly for you (might want to double check) if you have a `@Check` per precondition.

Reply:
Nope: if I write code like this

```
    @Check
    void check1() {
        throw new RuntimeException("I am sad");
    }

    @Check
    void check2() {
        throw new RuntimeException("I am very sad");
    }

    @Check
    void check3() {
        throw new RuntimeException("I am really sad");
    }
```

Immutables writes code like this. Fails on the first failing check (it happens to reverse the order), doesn't make any attempt to gather them together.

```
  private static ImmutableCsvSpecs validate(ImmutableCsvSpecs instance) {
    instance.check3();
    instance.check2();
    instance.check1();
    return instance;
  }
```